src/zap/zapimport.cpp

```diff
@@ -1773,6 +1774,45 @@ void ZapImportSectionSignatures::PlaceDynamicHelperCell(ZapImport * pImport)
     m_pImage->GetImportTable()->PlaceImportBlob(pCell);
 }
 
+ZapImport * ZapImportTable::GetDictionaryLookupCell(CORCOMPILE_FIXUP_BLOB_KIND kind, CORINFO_RESOLVED_TOKEN * pResolvedToken, CORINFO_LOOKUP * pLookup)
+{
+    _ASSERTE(pLookup->lookupKind.needsRuntimeLookup);
+
+    static DWORD counter = 0;
+
+    ZapImport * pImport = GetImport<ZapDynamicHelperCell, ZapNodeType_DynamicHelperCell>((void *)kind, (PVOID)(counter++));
+
+    if (!pImport->HasBlob())
+    {
+        SigBuilder sigBuilder;
+
+        sigBuilder.AppendData(kind & ~CORINFO_HELP_READYTORUN_ATYPICAL_CALLSITE);
+
+        // Write out key components of the CORINFO_RESOLVED_TOKEN structure used in the dictionary lookup:
+        //      1) encoded tokenContext
+        //      2) token
+        //      3) tokenType, coupled with information on the kind of resolved token handle
+
+        // tokenContext
+        GetCompileInfo()->EncodeMethod(m_pImage->GetModuleHandle(), pResolvedToken->tokenContext, &sigBuilder, NULL, NULL);
+
+        // token
+        // TODO: stop writing out token, and encode typespec/methodspec/hClass/hField/hMethod members
+        sigBuilder.AppendData(RidFromToken(pResolvedToken->token));
+        sigBuilder.AppendData(TypeFromToken(pResolvedToken->token) >> 24);
+
+        // token type
+        sigBuilder.AppendData(pResolvedToken->tokenType);
+
+        // Dictionary entry kind
```

Comment:
The encoding of dictionary entry kind should use values from the existing ReadyToRunFixupKind instead to keep the format sane.